### DISCUSSION TITLE
Keycloak in Istio

### DISCUSSION THREAD

**Discussion by jithkaringada-corp** [keycloak/keycloak#8864] [Q&A] [answered]
I have set up a local minikube in the hopes of replicating this set up in a production environment. I was able to use keycloak operator to set up a running pod with the istio-proxy sidecar container. However, 2 things came up. 
1. It looks like the keycloak container needs to terminate TLS. I was hoping to set up the keycloak container to listen on unencrypted port so that the sidecar terminated the TLS session. Is this possible?
2. If I am able to figure out how to get the sidecar container to not terminate the TLS, how do I update the client cert and key and CA chain using the keycloak operator. I get warnings that I am using the self generated certs.

**Comment by vmuzikar** [comment, marked as answer]
1. The operator currently doesn't support exposing KC over HTTP.
2. You can customise the TLS by mounting a cert and a key as [described in the container docs](https://github.com/keycloak/keycloak-containers/blob/main/server/README.md#setting-up-tlsssl). The operator doesn't verify the TLS certs when communicating with KC (unless `sso-x509-https-secret` secret is found but that's normally created only on OpenShift which I assume is not your case).

**Reply by jithkaringada-corp** [reply]
This is great info, thank you. I will resign to using standalone on a vm. It would be great to have keycloak work in a service mesh like istio where the TLS is terminated at the istio-proxy sidecar container. Will there be such a possibility with keycloak-x?

**Reply by mmosttler** [reply]
Is it possible for the Operator to have an option for the Service to expose port 8080?  Then you would have the open port and could possibly put it behind the istio proxy with mtls?

**Reply by jithkaringada-corp** [reply]
@vmuzikar I am not sure if there is an implementation with istio example out there, but in the new service mesh/microservices world (behind a reverse proxy set up), I could imagine this requirement growing in popularity. @mmosttler I couldn't see a way that port 8080 (straight http) could be set up with keycloak operator because it auto generated self signed certs, if you didn't explicitly set certs for it to create a keystore for you. I am interested in whether this kind of set up would be possible with keycloak-x.
